tests/lib/update_tpp_schema.py

```diff
@@ -0,0 +1,210 @@
+        {"ColumnName": "CodedEvent_ID", "ColumnType": "bigint"},
+        {"ColumnName": "NumericValue", "ColumnType": "real"},
+        {"ColumnName": "ConsultationDate", "ColumnType": "datetime"},
+        {"ColumnName": "ConceptID", "ColumnType": "varchar", "MaxLength": "18"},
+        {"ColumnName": "CodingSystem", "ColumnType": "int"},
+    ]
+
+    # We don't get column collation information but we know this matters in some cases
+    # because you can't compare columns across tables unless the collations are
+    # compatible. We add collations here for the two critical columns whose collations
+    # we're aware of (because they caused us problems in the distant past).
+    # https://github.com/opensafely/tpp-database-schema/issues/50
+    add_to_column(
+        by_table["CodedEvent"],
+        "CTV3Code",
+        collation="Latin1_General_BIN",
+    )
+    add_to_column(
+        by_table["MedicationDictionary"],
+        "DMD_ID",
+        collation="Latin1_General_CI_AS",
+    )
+
+
+def write_schema(lines):
+    lines[:0] = [HEADER.strip()]
+    code = "\n".join(lines)
+    code = black.format_str(code, mode=black.Mode())
+    SCHEMA_PYTHON.write_text(code)
+
+
+def sort_columns(columns):
+    # Assert column names are unique
+    assert len({c["ColumnName"] for c in columns}) == len(columns)
+    # Sort columns lexically except keep `Patient_ID` first
+    return sorted(
+        columns,
+        key=lambda c: (c["ColumnName"] != "Patient_ID", c["ColumnName"]),
+    )
+
+
+def add_to_column(columns, name, **kwargs):
+    for column in columns:
+        if column["ColumnName"] == name:
+            column.update(kwargs)
+            return
+    assert False, f"Column '{name}' not found"
+
+
+def class_name_for_table(name):
+    assert is_valid(name), name
+    return name
+
+
+def attr_name_for_column(name):
+    name = name.replace(".", "_")
+    if name == "class":
+        name = "class_"
+    assert is_valid(name), name
+    return name
+
+
+def definition_for_column(column):
+    type_formatter = TYPE_MAP[column["ColumnType"]]
+    args = [type_formatter(column)]
+    # If the name isn't a valid Python attribute then we need to supply it explicitly as
+    # the first argument
+    name = column["ColumnName"]
+    if attr_name_for_column(name) != name:
+        args.insert(0, repr(name))
+    return f"mapped_column({', '.join(args)})"
+
+
+def format_string_type(type_name, column):
+    args = column["MaxLength"]
+    if collation := column.get("collation"):
+        args += f", collation={collation!r}"
+    return f"{type_name}({args})"
+
+
+def is_valid(name):
+    return name.isidentifier() and not keyword.iskeyword(name)
+
+
+if __name__ == "__main__":
+    command = sys.argv[1] if len(sys.argv) > 1 else None
+    if command == "fetch":
+        fetch_schema()
+    elif command == "build":
+        build_schema()
+    else:
+        raise RuntimeError(f"Unknown command: {command}")
```

Comment:
suggestion: maybe have a more helpful error here, so someone doesn't have to read the source to figure out the script?

Reply:
Maybe even something terse like:

```suggestion
        raise RuntimeError(f"Unknown command: {command}; valid commands: fetch, build")
```